secondLayout.js

```diff
@@ -0,0 +1,317 @@
+// objects overlapping
+function crashesWithAnything(obj) {
+  for (i = 0; i < objectsArr.length; i++) {
+    if (obj.crash(objectsArr[i])) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// draw function for the page after winning or losing the game
+let drawEndGame = () => {
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  // draw the bg
+  // draw img if the player won
+  if (!runningGame && !winGame) {
+    drawEndGameBg();
+    let winImg = new Image();
+    winImg.src = "images/You-win.png";
+    ctx.drawImage(winImg, 400, 300, 400, 50);
+    // draw another img if the player lost
+  } else {
+    bodyCtx.style.backgroundImage =
+      "url('images/stars.png'),url('images/endGameBg/background.png')";
+    drawEndGameBg();
+    let winImg = new Image();
+    winImg.src = "images/lose.png";
+    ctx.drawImage(winImg, 400, 300, 400, 50);
+  }
+  // press space to play the game again
+
+  window.requestAnimationFrame(drawEndGame);
+};
+
+// For ending the game
+let runningGame = true;
+
+// for winning the game
+let winGame = true;
+// to stop the typing sound from playing
+let isOutOfFirstPage = false;
+//Animation frames
+let animationFrame = 0;
+
+// declare the counter
+let counter = 0;
+
+// declare an objects array for all objects
+let objectsArr = [];
+
+// declare an object for the score of each category
+let scores = {
+  money: 0,
+  health: 0,
+  entertainment: 0,
+  love: 0
+};
+
+// function to draw the game on canvas
+let drawGameBoard = () => {
+  if (!runningGame) {
+    drawEndGame();
+    gameSound.pause();
+    return;
+  }
+
+  gameSound.volume = 0.1;
+  gameSound.play();
+  // clear the canvas
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  // increase the counter
+  counter++;
+
+  //   draw background image
+  drawGameBoardBg();
+
+  // draw the player
+  player.update();
+  // and add the score for the object category.
+  let ind;
+  objectsArr.forEach((e, i) => {
+    if (player.crash(e)) {
+      // add the score
+      if (scores[e.scoreType] <= 9) {
+        scores[e.scoreType] += 1;
+      }
+      // get the index of the object in the array after the play with it
+      ind = i;
+      // play the sound depending on the score type
+      if (e.scoreType == "money") {
+        moneySound.play();
+      }
+      if (e.scoreType == "health") {
+        healthSound.play();
+      }
+      if (e.scoreType == "entertainment") {
+        entertainmentSound.play();
+      }
+      if (e.scoreType == "love") {
+        loveSound.play();
+      }
+
+      if (e.scoreType == "obstacles") {
+        runningGame = false;
+        gameOverSound.play();
+      }
+    }
+
+
+    // draw the object in the array
+    if (e.scoreType == "obstacles") {
+      //Animation frame counter
+
+      if (counter % 8 === 0) {
+        //To have the acces every %8 frame and update the index in the class obstacle
+        e.currentImageIdx = (e.currentImageIdx + 1) % e.images.length;
+      }
+      e.update();
```

Comment:
since `e.update()` is called in both options of the if clause it should simply be outside the if clause